src/dmd/dsymbolsem.d

```diff
@@ -1570,7 +1593,7 @@ extern(C++) final class Semantic3Visitor : Visitor
                 }
 
                 // If declaration has no body, don't set sbody to prevent incorrect codegen.
-                if (funcdecl.fbody || funcdecl.fdensure || funcdecl.fdrequire)
+                if (funcdecl.fbody || funcdecl.allowsContractWithoutBody())
```

Comment:
The test of `funcdecl.fbody` here is redundant with what `allowsContractWithoutBody` does.

Reply:
I've turned it into an assert in the function.